components/script/dom/htmlselectelement.rs

```diff
@@ -183,6 +184,65 @@ impl HTMLSelectElementMethods for HTMLSelectElement {
     fn Labels(&self) -> Root<NodeList> {
         self.upcast::<HTMLElement>().labels()
     }
+
+    // https://html.spec.whatwg.org/multipage/#dom-select-length
+    fn SetLength(&self, value: u32) {
+        let length  = self.Length();
+        let node = self.upcast::<Node>();
+        if value < length {     // truncate the number of option elements
+            for _ in 0..(length - value) {
+                let lastChild = node.GetLastChild();
```

Comment:
I think we can just use `node.descending_last_children().take(length - value)` here.